pkg/services/ngalert/api/authorization_test.go

```diff
@@ -0,0 +1,56 @@
+package api
+
+import (
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/go-openapi/loads"
+	"github.com/stretchr/testify/require"
+
+	acmock "github.com/grafana/grafana/pkg/services/accesscontrol/mock"
+)
+
+func TestAuthorize(t *testing.T) {
+	json, err := os.ReadFile(filepath.Join("tooling", "spec.json"))
+	require.NoError(t, err)
+	swaggerSpec, err := loads.Analyzed(json, "")
+	require.NoError(t, err)
+
+	paths := make(map[string][]string)
+
+	for p, item := range swaggerSpec.Spec().Paths.Paths {
+		var methods []string
+
+		if item.Get != nil {
+			methods = append(methods, http.MethodGet)
+		}
+		if item.Put != nil {
+			methods = append(methods, http.MethodPut)
+		}
+		if item.Post != nil {
+			methods = append(methods, http.MethodPost)
+		}
+		if item.Delete != nil {
+			methods = append(methods, http.MethodDelete)
+		}
+		if item.Patch != nil {
+			methods = append(methods, http.MethodPatch)
+		}
+		paths[p] = methods
+	}
+
+	require.NoErrorf(t, err, "failed to read swagger specification")
+
+	ac := acmock.New()
+	api := &API{AccessControl: ac}
+
```

Comment:
Can we please assert the number of currently expected paths? I think it would a) avoid someone removing all the paths and this test passing b) having to manually increase the counter if you add/remove a path (and thus be aware that you need to change the permission model.